test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+        [
+          authorizationIds[0],
+          createBytesString(payments[payments.length - 1].authorizationId + 1, BYTES16_LENGTH)
+        ]
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_DOES_NOT_EXIST);
+    });
+
+    it("Is reverted if one of the payments is uncleared", async () => {
+      await proveTx(cardPaymentProcessor.connect(admin).unclearPayment(
+        authorizationIds[0]
+      ));
+      await expect(cardPaymentProcessor.connect(admin).unclearPayments(
+        authorizationIds
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_IS_UNCLEARED);
+    });
+
+    it("Does not transfer tokens, emits the correct events, changes the state properly", async () => {
+      await checkCardPaymentProcessorState(payments);
+      await expect(cardPaymentProcessor.connect(admin).unclearPayments(
+        authorizationIds
+      )).to.changeTokenBalances(
+        brlcMock,
+        [cardPaymentProcessor, ...accountAddresses],
+        [0, ...accountAddresses.map(() => 0)]
+      ).and.to.emit(
+        cardPaymentProcessor,
+        "UnclearPayment"
+      ).withArgs(
+        authorizationIds[0],
+        payments[0].account.address,
+        payments[0].amount,
+        expectedClearedBalances[0],
+        expectedUnclearedBalances[0],
+        payments[0].revocationCounter || 0
+      ).and.to.emit(
+        cardPaymentProcessor,
+        "UnclearPayment"
+      ).withArgs(
+        authorizationIds[1],
+        payments[1].account.address,
+        payments[1].amount,
+        expectedClearedBalances[1],
+        expectedUnclearedBalances[1],
+        payments[1].revocationCounter || 0
+      );
+      payments.forEach(payment => payment.status = PaymentStatus.Uncleared)
+      await checkCardPaymentProcessorState(payments);
+    });
+  });
+
+  describe("Function 'revokePayment()'", async () => {
+    const reversingPaymentCorrelationId: string = SOME_BYTES16_STRING;
+    const parentTxHash: string = SOME_BYTES32_STRING;
+
+    let payment: TestPayment;
+    let authorizationId: string;
+    let admin: SignerWithAddress;
+
+    beforeEach(async () => {
+      payment = {
+        authorizationId: 987,
+        account: user1,
+        amount: 876,
+        status: PaymentStatus.Nonexistent,
+        revocationCounter: 0,
+        makingPaymentCorrelationId: 765,
+        parentTxHash: parentTxHash
+      };
+      admin = user2;
+      authorizationId = createBytesString(payment.authorizationId, BYTES16_LENGTH);
+      await setUpContractsForPayments([payment]);
+      await makePayments([payment]);
+      await addAccountToCardPaymentProcessorWhitelist(admin);
+    });
+
+    it("Is reverted if the contract is paused", async () => {
+      await proveTx(cardPaymentProcessor.setPauser(deployer.address));
+      await proveTx(cardPaymentProcessor.pause());
+      await expect(cardPaymentProcessor.connect(admin).revokePayment(
+        authorizationId,
+        reversingPaymentCorrelationId,
```

Comment:
Codacy has a fix for the issue: [Insert `····`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
            reversingPaymentCorrelationId,
```